galley/pkg/config/source/kube/apiserver/source.go

```diff
@@ -118,6 +117,7 @@ func (s *Source) Start() {
 		key := asKey(r.Group, r.Kind)
 		s.expectedResources[key] = r
 	}
+	s.mu.Unlock()
```

Comment:
I know we've had some trouble around this lock in the past, it may be worth adding a comment that the reason we don't do this at the *very* end is to avoid a deadlock with crdWatcher after it starts since crdWatcher also tries to grab the lock. 

At least, that's what I remember as the reason why we don't just unlock at the end of the function.

Reply:
Good call. Done.